### PR TITLE
Add fu_udev_device_get_parent_with_subsystem_devtype() for future use

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupdplugin/fu-udev-device.c [hunk 1]
```diff
@@ -1259,23 +1259,62 @@ fu_udev_device_match_subsystem_devtype(GUdevDevice *udev_device,
...
-static GUdevDevice *
-fu_udev_device_get_parent_with_subsystem_devtype(GUdevDevice *udev_device,
+/**
+ * fu_udev_device_get_parent_with_subsystem_devtype
+ * @self: a #FuUdevDevice
+ * @subsystem: (nullable): the name of a udev subsystem
+ * @devtype: (nullable): the name of a udev devtype, e.g. `usb_interface`
+ * @error: (nullable): optional return location for an error
+ *
+ * Get the device that is a parent of self and has the provided subsystem and device type.
```
Comment (superm1) [inline libfwupdplugin/fu-udev-device.c:1271]: It keeps going up right?
I think it should be ancestor.

Comment (hughsie) [reply libfwupdplugin/fu-udev-device.c:1271]: Agreed, fixed. Thanks.

File: libfwupdplugin/fu-udev-device.c [hunk 2]
```diff
@@ -1259,23 +1259,62 @@ fu_udev_device_match_subsystem_devtype(GUdevDevice *udev_device,
...
+ * @error: (nullable): optional return location for an error
+ *
+ * Get the device that is a ancestor of self and has the provided subsystem and device type.
+ *
+ * Returns: (transfer full): device, or %NULL
+ *
+ * Since: 2.0.0
+ */
+FuUdevDevice *
+fu_udev_device_get_parent_with_subsystem_devtype(FuUdevDevice *self,
```
Comment (superm1) [inline libfwupdplugin/fu-udev-device.c:1278]: sorry I should have made my comment clearer.  Not just the comment needs updating, I was suggesting the function too.

IE `fu_udev_device_get_ancestor_with_subsystem_devtype`

Comment (hughsie) [reply libfwupdplugin/fu-udev-device.c:1278]: I'm not sure ancestor works -- it makes it sound like it's an old device that's no longer here. Counterproposal -- we keep the existing `fu_udev_device_get_parent_with_subsystem()` and just add the `devtype` to that-- with `NULL` being "don't care"?

Comment (hughsie) [reply libfwupdplugin/fu-udev-device.c]: narrator: he did that, and liked it.

**Review by superm1 — APPROVED** [review]

**Pull request merged**